apps/rewards/src/invite/inviteReward.service.ts

```diff
@@ -0,0 +1,171 @@
+    private readonly logger: KomenciLoggerService
+  ) {
+    this.komenciAddresses = this.networkCfg.relayers.map(
+      relayer => relayer.externalAccount
+    )
+  }
+
+  @Cron(CronExpression.EVERY_MINUTE)
+  async sendInviteRewards() {
+    if (this.isRunning) {
+      this.logger.log('Skipping because previous run is still ongoing')
+      return
+    }
+    this.isRunning = true
+
+    const cUsdTokenAddress = (
+      await this.contractKit.registry.addressFor(CeloContract.StableToken)
+    ).toLowerCase()
+
+    await this.usingLastNotifiedBlock(async fromBlock => {
+      this.logger.log(
+        `Starting to fetch invite rewards from block ${fromBlock}`
+      )
+      const escrow = await this.contractKit.contracts.getEscrow()
+      const withdrawalEvents = await escrow.getPastEvents('Withdrawal', {
+        fromBlock
+      })
+      this.logger.log(`Withdrawal events received: ${withdrawalEvents.length}`)
+      let maxBlock = fromBlock
+      for (const withdrawal of withdrawalEvents) {
+        const {
+          transactionHash,
+          blockNumber,
+          returnValues: { identifier, token, to }
+        } = withdrawal
+        maxBlock = Math.max(maxBlock, blockNumber)
+
+        const inviter = to.toLowerCase()
+        if (cUsdTokenAddress !== token.toLowerCase()) {
+          continue
+        }
+        const tx = await this.contractKit.web3.eth.getTransaction(
+          transactionHash
+        )
+        if (!this.isKomenciSender(tx.from)) {
+          continue
+        }
+        const invitee = tx.to?.toLowerCase() ?? ''
+        const conditions = await Promise.all([
+          this.addressIsConsideredVerified(inviter, identifier),
+          this.addressIsConsideredVerified(invitee, identifier),
+          this.inviterHasNotReachedWeeklyLimit(inviter),
+          this.inviteeRewardNotInProgress(invitee)
+        ])
+        if (conditions.every(condition => condition)) {
+          this.logger.log(
+            `Starting to send reward to ${inviter} for inviting ${invitee}`
+          )
+          try {
+            const inviteReward = await this.createInviteReward(inviter, invitee)
+            this.sendInviteReward(inviteReward)
+          } catch (error) {
+            this.logger.log(`Error creating and sending reward ${error}`)
+          }
+        }
+      }
+      return maxBlock
+    })
+
+    this.isRunning = false
+  }
+
+  async usingLastNotifiedBlock(fn: (blockNumber: number) => Promise<number>) {
+    try {
+      let lastNotifiedBlock = await this.notifiedBlockRepository.findOne({
+        key: 'inviteReward'
+      })
+      if (!lastNotifiedBlock) {
+        lastNotifiedBlock = {
+          id: uuidv4(),
+          key: 'inviteReward',
+          // TODO: Replace this by the correct starting block in alfajores and mainnet.
+          // For now, putting a high value to avoid potential issues if this is accidentally deployed.
+          blockNumber: 100000000
```

Comment:
Should we set this to be the current highest block? This would protect us from repaying everyone if the DB mistakenly is wiped